src/mqt/predictor/ml/predictor.py

```diff
@@ -356,171 +343,89 @@ def get_prepared_training_data(
+        with resources.as_file(ml.helper.get_path_training_data() / "training_data_aggregated") as path:
+            data = np.asarray(training_data, dtype=object)
+            np.save(str(path / ("training_data_" + self.figure_of_merit + ".npy")), data)
+            data = np.asarray(names_list, dtype=str)
+            np.save(str(path / ("names_list_" + self.figure_of_merit + ".npy")), data)
+            data = np.asarray(scores_list, dtype=object)
+            np.save(str(path / ("scores_list_" + self.figure_of_merit + ".npy")), data)
 
-        plt.ylim(0, 1.05)
-        plt.xlim(0, len(scores_filtered))
-        result_path = Path("results")
-        if not result_path.is_dir():
-            result_path.mkdir()
-        plt.savefig(result_path / "y_pred_eval_normed.pdf", bbox_inches="tight")
-
-    def predict_probs(self, qc: Path | QuantumCircuit, figure_of_merit: reward.figure_of_merit) -> NDArray[np.float64]:
-        """Returns the probabilities for all supported quantum devices to be the most suitable one for the given quantum circuit.
+    def load_training_data(self) -> tuple[list[NDArray[np.float64]], list[str], list[NDArray[np.float64]]]:
+        """Loads and returns the training data from the training data folder.
 
         Arguments:
-            qc: The QuantumCircuit or Path to the respective qasm file.
-            figure_of_merit: The figure of merit to be used for prediction.
+            figure_of_merit: The figure of merit to be used for compilation. Defaults to "expected_fidelity".
 
         Returns:
-            The probabilities for all supported quantum devices to be the most suitable one for the given quantum circuit.
+           The training data, the names list and the scores list.
         """
-        if self.clf is None:
-            path = ml.helper.get_path_trained_model(figure_of_merit)
-            if path.is_file():
-                self.clf = load(path)
-
-            if self.clf is None:
-                error_msg = "The ML model is not trained yet. Please train the model before using it."
-                logger.error(error_msg)
+        with resources.as_file(ml.helper.get_path_training_data() / "training_data_aggregated") as path:
+            if (
+                path.joinpath("training_data_" + self.figure_of_merit + ".npy").is_file()
+                and path.joinpath("names_list_" + self.figure_of_merit + ".npy").is_file()
+                and path.joinpath("scores_list_" + self.figure_of_merit + ".npy").is_file()
+            ):
+                training_data = np.load(path / ("training_data_" + self.figure_of_merit + ".npy"), allow_pickle=True)
+                names_list = list(np.load(path / ("names_list_" + self.figure_of_merit + ".npy"), allow_pickle=True))
+                scores_list = list(np.load(path / ("scores_list_" + self.figure_of_merit + ".npy"), allow_pickle=True))
+            else:
+                error_msg = "Training data not found. Please run the training script first."
```

Comment:
might be worth adding a test covering this error case as it is something that people might actually be hitting.

Reply:
I improved the message and added a test to cover it.